File: backend/api/performance.py
from fastapi import APIRouter, HTTPException
from .models import PerformanceData
import time

router = APIRouter()

# Mock data that matches the TypeScript mockPerformanceData
mock_performance_data = {
    "winRate": 68.5,
    "avgPnl": 2.4,
    "totalTrades": 124,
    "monthlyPnl": 18.7,
    "pnlHistory": [
        {"date": "2023-06-01", "pnl": 4.33},
        {"date": "2023-06-02", "pnl": 8.2},
        {"date": "2023-06-03", "pnl": 6.85},
        {"date": "2023-06-04", "pnl": 6.85},
        {"date": "2023-06-05", "pnl": 4.97},
        {"date": "2023-06-06", "pnl": 7.25},
        {"date": "2023-06-07", "pnl": 10.5},
        {"date": "2023-06-08", "pnl": 12.8},
        {"date": "2023-06-09", "pnl": 11.2},
        {"date": "2023-06-10", "pnl": 14.5},
        {"date": "2023-06-11", "pnl": 13.8},
        {"date": "2023-06-12", "pnl": 16.2},
        {"date": "2023-06-13", "pnl": 15.4},
        {"date": "2023-06-14", "pnl": 18.7},
    ]
}

@router.get("/performance", response_model=PerformanceData)
async def get_performance():
    # Simulate API delay
    time.sleep(0.7)
    
    return mock_performance_data

File: src/api/performance.ts
// Define PerformanceData interface locally instead of importing from mockApi
export interface PerformanceData {
  winRate: number;
  totalTrades: number;
  avgPnl: number;
  monthlyPnl: number;
  pnlHistory: { date: string; value: number }[];
}

// Default performance data when live data is unavailable
const defaultPerformanceData: PerformanceData = {
  winRate: 68,
  totalTrades: 42,
  avgPnl: 2.7,
  monthlyPnl: 8.4,
  pnlHistory: [
    { date: "2023-06-01", value: 1.2 },
    { date: "2023-06-15", value: 3.5 },
    { date: "2023-07-01", value: 5.8 },
    { date: "2023-07-15", value: 8.4 },
  ],
};

// Fetch performance metrics
export async function getPerformance(): Promise<PerformanceData> {
  console.log("[API] Fetching performance data...");
  try {
    const response = await fetch("/api/performance");

    if (!response.ok) {
      console.log(
        `[API] Performance data fetch failed: ${response.status} ${response.statusText}`,
      );
      console.log("[API] Returning default performance data");
      return defaultPerformanceData;
    }

    const data = await response.json();
    console.log("[API] Performance data fetched successfully:", data);
    return data;
  } catch (error) {
    console.error("Failed to fetch performance data:", error);
    console.log("[API] Returning default performance data due to error");
    return defaultPerformanceData;
  }
}
